queue/src/main/java/org/killbill/queue/DefaultQueueLifecycle.java

```diff
@@ -152,11 +197,17 @@ private void sleepALittle(final long loopTimeMsec) throws InterruptedException {
     public void stopQueue() {
         this.isProcessingEvents = false;
 
-        executor.shutdownNow();
+        executor.shutdown();
         try {
             executor.awaitTermination(config.getPollingSleepTimeMs(), TimeUnit.SECONDS);
         } catch (final InterruptedException e) {
             log.info(String.format("%s: Stop sequence has been interrupted", svcQName));
+        } finally {
+            int remainingCompleted = completedOrFailedEvents.size();
+            int remainingRetried = retriedEvents.size();
+            if (remainingCompleted > 0 || remainingRetried > 0) {
+                log.warn(String.format("%s: Stopped queue with %d event/notifications non completed ", svcQName, (remainingCompleted + remainingRetried)));
```

Comment:
Nit: no need for the `String.format`.

Reply:
I started to make the change, but i realized it's all over this code - is this worth it?